backend/src/test/java/de/laura/project/controller/TriviaControllerTest.java

```diff
@@ -39,45 +45,104 @@ private String getUrl() {
-        answerForTriviaQuestionSet.add("correct answer");
-        answerForTriviaQuestionSet.add("answerOne");
-        answerForTriviaQuestionSet.add("answerTwo");
-        answerForTriviaQuestionSet.add("answerThree");
-
-        ArrayList<String> anotherAnswerForTriviaQuestionSet = new ArrayList<>();
-        answerForTriviaQuestionSet.add("another correct answer");
-        answerForTriviaQuestionSet.add("answerOne");
-        answerForTriviaQuestionSet.add("answerTwo");
-        answerForTriviaQuestionSet.add("answerThree");
+        String ApiUrl = "https://opentdb.com/api.php?amount=10&category=12&difficulty=medium&type=multiple";
 
         List<String> answersForApiDataList = List.of("answerOne", "answerTwo", "answerThree");
 
-        List<TriviaApiData> mockedTriviaApiDataList = List.of(
-                new TriviaApiData("12", "multiple choice", "easy", "This is an awesome question", "correct answer", answersForApiDataList),
-                new TriviaApiData("12", "multiple choice", "easy", "This is another awesome question", "another correct answer", answersForApiDataList));
+        ArrayList<TriviaApiData> mockedTriviaApiDataList = new ArrayList<>();
+        mockedTriviaApiDataList.add(new TriviaApiData("12", "multiple choice", "easy", "This is an awesome question", "correct answer", answersForApiDataList));
+        mockedTriviaApiDataList.add(new TriviaApiData("12", "multiple choice", "easy", "This is another awesome question", "another correct answer", answersForApiDataList));
 
         TriviaApiDataAggregation mockedTriviaDataAggregationList = new TriviaApiDataAggregation(1, mockedTriviaApiDataList);
 
-        when(restTemplate.getForEntity(url, TriviaApiDataAggregation.class))
+        when(restTemplate.getForEntity(ApiUrl, TriviaApiDataAggregation.class))
                 .thenReturn(new ResponseEntity<>(mockedTriviaDataAggregationList, HttpStatus.OK));
 
+        TriviaQuestionSetDTO triviaQuestionSetDTO = new TriviaQuestionSetDTO(10, 12, "medium");
+
+        //WHEN
+        ResponseEntity<Void> response = testRestTemplate.postForEntity(getUrl(), triviaQuestionSetDTO, Void.class);
+
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+    }
+
+
+    @Test
+    @DisplayName("Checks whether user's answer is correct")
+
+    public void checkAnswerIsCorrect() {
+
+        //GIVEN
+        List<String> answerForTriviaQuestionSet = List.of("correct answer", "answerOne", "answerTwo", "answerThree");
+        List<String> anotherAnswerForTriviaQuestionSet = List.of("another correct answer", "answerOne", "answerTwo", "answerThree");
+
+        tempTriviaQuestionDB.getTriviaQuestionSetList().add(new TriviaQuestionSet(1, "This is an awesome question", "correct answer", answerForTriviaQuestionSet));
+        tempTriviaQuestionDB.getTriviaQuestionSetList().add(new TriviaQuestionSet(2, "This is another awesome question", "another correct answer", anotherAnswerForTriviaQuestionSet));
```

Comment:
if you do not need this data point for your test you can also delete it because it is not used anywhere